src/services/ecommerce/payments/hooks.ts

```diff
@@ -0,0 +1,100 @@
+import { useQuery, useQueryClient, useMutation } from "@tanstack/react-query";
+import { paymentsApi } from "../client";
+import type {
+  PaymentsApiPaymentsBasketsListRequest,
+  PaymentsApiPaymentsBasketsCreateFromProductCreateRequest,
+  PaymentsApiPaymentsBasketsAddDiscountCreateRequest,
+  PaymentsApiPaymentsCheckoutCreateRequest,
+} from "../generated/v0/api";
+
+const usePaymentsBasketList = (
+  options: PaymentsApiPaymentsBasketsListRequest,
+) =>
+  useQuery({
+    queryKey: ["paymentsBaskets"],
```

Comment:
```suggestion
    queryKey: ["paymentsBaskets", options],
```
Anything that's used in the queryFn should go in the queryKey, otherwise

```
you make request with ?limit=10
later you make request with ?limit=20, and get cached value from earlier
```